src/caffe/layers/pooling_layer.cu

```diff
@@ -149,7 +149,7 @@ __global__ void StoPoolForwardTest(const int nthreads,
         cumvalues += bottom_slice[h * width + w] * bottom_slice[h * width + w];
       }
     }
-    top_data[index] = cumvalues / cumsum;
+    top_data[index] = (cumsum > 0.) ? cumvalues / cumsum : 0.;
```

Comment:
Negative values aren't valid for stochastic pooling.  The input represents values in a probability distribution (though it doesn't have to be normalized to sum to 1).